src/StatusTable/StatusRow.tsx

```diff
@@ -20,15 +20,27 @@ export interface StatusRowProps {
      * This is to set an external classname.
      */
     className?: string;
+
+    /**
+     * This is to handle the row click event.
+     */
+    onRowClick?: () => void;
 }
 
 /**
  *  Status Row Component is used to render the row with various colums.
  */
-const StatusRow: FC<StatusRowProps> = ({ children, status = Status.NONE, className }) => {
+const StatusRow: FC<StatusRowProps> = ({ children, status = Status.NONE, className, onRowClick }) => {
+
+    const handleOnClick = () => {
+        if (onRowClick) {
+            onRowClick();
+        }
+    }
+
     return (
         <ThemeProvider theme={theme}>
-            <StatusRowStyle className={className} status={status}>
+            <StatusRowStyle className={className} status={status} rowClick={Boolean(onRowClick)} onClick={() => handleOnClick()}>
```

Comment:
This could be simplified to:

` <StatusRowStyle className={className} status={status} rowClick={Boolean(onRowClick)} onClick={handleOnClick}>`